bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java

```diff
@@ -133,9 +134,136 @@ public static RrdDbPool getDatabasePool() {
         return DATABASE_POOL;
     }
 
+    private final ScheduledFuture<?> storeJob;
+
     @Activate
     public RRD4jPersistenceService(final @Reference ItemRegistry itemRegistry) {
         this.itemRegistry = itemRegistry;
+        storeJob = scheduler.scheduleWithFixedDelay(() -> doStore(false), 1, 1, TimeUnit.SECONDS);
```

Comment:
Wouldn't this line better be placed in the activate() method behind `active = true`? Then there wouldn't be the need to set a 1 second delay either.

Reply:
Yes, that should work as well.